server/src/main/java/org/elasticsearch/repositories/blobstore/ChecksumBlobStoreFormat.java

```diff
@@ -112,23 +101,16 @@ public ChecksumBlobStoreFormat(String codec, String blobNameFormat, CheckedFunct
      * @param name          name to be translated into
      * @return parsed blob object
      */
-    public T read(BlobContainer blobContainer, String name) throws IOException {
+    public T read(BlobContainer blobContainer, String name, NamedXContentRegistry namedXContentRegistry) throws IOException {
         String blobName = blobName(name);
-        return readBlob(blobContainer, blobName);
+        return deserialize(blobName, namedXContentRegistry, Streams.readFully(blobContainer.readBlob(blobName)));
     }
 
     public String blobName(String name) {
         return String.format(Locale.ROOT, blobNameFormat, name);
     }
 
-    /**
-     * Reads blob with specified name without resolving the blobName using using {@link #blobName} method.
-     *
-     * @param blobContainer blob container
-     * @param blobName blob name
-     */
-    public T readBlob(BlobContainer blobContainer, String blobName) throws IOException {
-        final BytesReference bytes = Streams.readFully(blobContainer.readBlob(blobName));
+    public T deserialize(String blobName, NamedXContentRegistry namedXContentRegistry, BytesReference bytes) throws IOException {
```

Comment:
Exposing this here for the mock repo tests for now but this is needed for the repair logic as well.